### PR TITLE
Support direct methods for Azure IoT Hub Service

### PR DESCRIPTION
**Pull request by corollaries**
For a couple of months I've been working on creating a wrapper for the Azure IoT Hub Service SDK as a personal project to learn Rust. Meanwhile I've been keeping an eye on the `azure-sdk-for-rust` repository as well. Found some similarities and have spent a couple of hours to merge some of the functionality that I've implemented into this repository. Maybe this could be useful for others!

## Features
This PR would add support for:
- Creating an Azure IoT Hub ServiceClient from either:
  - A given connection string from one of the shared access policies.
  - A private key from one of the shared access policies.
  - A previously generated shared access token.
- Creating a direct method that can be invoked on an IoT Hub Device or an IoT Edge module.

## Thoughts
- The example and code have only been tested on a couple of Linux distributions. If need be, I'd be able to test it on Windows as well.
- I noticed by sifting through the already existing code that `hyper-rustls` is being used for hyper TLS support. Unfortunately after some testing I did not manage to get the `ServiceClient` to work with this due to some server errors. Did not investigate further however and continued using `hyper-tls` for now, though it did seem to be some problem with `hyper-rustls` itself. 
- There are two other features that could be merged in the near future if this PR would be found acceptable. Both are already implemented but would need some porting from one repository to the other. I decided to keep this initial PR a bit smaller for now however. Two other features are:
  - Support for device and module twins. 
  - Setting the Azure IoT Edge modules configuration.

### CODE REVIEW

File: sdk/iothub/src/service/mod.rs [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+        let key = match decode(private_key) {
+            Ok(val) => val,
+            Err(err) => {
+                return Err(AzureError::GenericErrorWithText(format!(
+                    "Failed to decode the given private key: {}",
+                    err.to_string()
+                )));
+            }
+        };
```
Comment (rylev) [inline sdk/iothub/src/service/mod.rs:70]:
```suggestion
        let key = decode(private_key).map_err(|err|
            AzureError::GenericErrorWithText(format!(
                    "Failed to decode the given private key: {}",
                    err.to_string()
       )))?;
```

File: sdk/iothub/src/service/mod.rs [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+        let mut hmac = match HmacSHA256::new_varkey(key.as_ref()) {
```
Comment (rylev) [inline sdk/iothub/src/service/mod.rs:72]: Same suggestion as above ^^

**Review by rylev — COMMENTED** [review]
Looks good to me. Just some nitpicks.

I don't have experience with the iothubs SDK. @MindFlavor, do you mind looking at this as well?

**Review by MindFlavor — APPROVED** [review]
I'm not an expert either but it seems fine to me as well. 

The only thing I would like to address - but at a later stage - is to try and integrate better the `generate_sas_token` function in the core crate (to avoid replicating the same logic).

Great work BTW!

**Pull request merged**